### PR TITLE
Documenting the module a bit.

### PR DESCRIPTION
**Pull request by bartavelle**
I tried documenting the module a little bit, including an usage example.

I also tried highlighting the bad practice of collecting all file content before processing it.

Do you have suggestions about the content of this commit ? I will start publishing the package as soon as this is merged.

### CODE REVIEW

**Comment by snoyberg** [comment]
Signing off for the weekend, I'll review next week. Looking forward to it 👍

File: src/Data/Conduit/Tar.hs [hunk 1]
```diff
@@ -203,6 +210,41 @@ withEntry inner = do
             Just (ChunkException e) -> throwM e
             Nothing -> return ()
 
+{-| This function handles each entry of the tar archive according to the
+behaviour of the function passed as first argument.
+
+Here is a full example function, that read a compressed tar archive and for each entry that is a simple file, it returns its file path and SHA256 digest. Note that function can throw exceptions!
```
Comment (snoyberg) [inline src/Data/Conduit/Tar.hs]: s/read/reads

File: src/Data/Conduit/Tar.hs [hunk 2]
```diff
@@ -203,6 +210,41 @@ withEntry inner = do
...
+>                                .| CT.untar                 -- decode the tar archive
+>                                .| CT.withEntries hashentry -- process each file
+>                                .| sinkList                 -- collect the results
+>                                )
+>     where
+>         hashentry hdr = when (CT.headerFileType hdr == CT.FTNormal) $ do
+>             hash <- CH.sinkHash
+>             yield (CT.headerFilePath hdr, hash)
+
+The @hashentry@ function handles on a single entry, based on its first 'Header' argument.
```
Comment (snoyberg) [inline src/Data/Conduit/Tar.hs]: s/handles on a/handles a

File: src/Data/Conduit/Tar.hs [hunk 3]
```diff
@@ -203,6 +210,41 @@ withEntry inner = do
...
+In this example, a 'Consumer' is used to process the whole entry.
+
+Note that the benefits of stream processing are easily lost when working with a 'Consumer'. For example, the following implementation can use an unbounded amount of memory:
```
Comment (snoyberg) [inline src/Data/Conduit/Tar.hs]: Very nice :)

File: src/Data/Conduit/Tar.hs [hunk 4]
```diff
@@ -203,6 +210,41 @@ withEntry inner = do
...
+> import Control.Monad (when)
+> import Data.Conduit.Zlib (ungzip)
+> import Data.ByteString (ByteString)
+> 
+> filedigests :: FilePath -> IO [(FilePath, Digest SHA256)]
+> filedigests fp = runConduitRes (  sourceFileBS fp          -- read the raw file
+>                                .| ungzip                   -- gunzip
+>                                .| CT.untar                 -- decode the tar archive
+>                                .| CT.withEntries hashentry -- process each file
+>                                .| sinkList                 -- collect the results
```
Comment (snoyberg) [inline src/Data/Conduit/Tar.hs]: Perhaps worth instead using `mapM_C print` or `printC` to avoid holding all of the filepaths and hashes in memory.

**Comment by bartavelle** [comment]
I changed the commit, taking into account your remarks. Thanks for the review!

**Comment by snoyberg** [comment]
Looking good!

**Pull request merged**